### ISSUE TITLE
"Unknown Revision"

### ISSUE THREAD

**Issue by jherdman** [kellyselden/ember-cli-update#248]
I'm trying to upgrade to 2.18.1 this morning, and I'm seeing the following:

```
$ DEBUG=ember-cli-update,git-diff-apply ember-cli-update
  ember-cli-update npm info ember-cli versions --json +0ms
  ember-cli-update [
  ember-cli-update   "0.0.0",
  ember-cli-update   "0.0.1",
  ember-cli-update   "0.0.2",
  ember-cli-update   "0.0.3",
  ember-cli-update   "0.0.4",
  ember-cli-update   "0.0.5",
  ember-cli-update   "0.0.6",
  ember-cli-update   "0.0.7",
  ember-cli-update   "0.0.8",
  ember-cli-update   "0.0.9",
  ember-cli-update   "0.0.10",
  ember-cli-update   "0.0.11",
  ember-cli-update   "0.0.12",
  ember-cli-update   "0.0.13",
  ember-cli-update   "0.0.14",
  ember-cli-update   "0.0.15",
  ember-cli-update   "0.0.16",
  ember-cli-update   "0.0.18",
  ember-cli-update   "0.0.19",
  ember-cli-update   "0.0.20",
  ember-cli-update   "0.0.21",
  ember-cli-update   "0.0.22",
  ember-cli-update   "0.0.23",
  ember-cli-update   "0.0.24",
  ember-cli-update   "0.0.25",
  ember-cli-update   "0.0.27",
  ember-cli-update   "0.0.28",
  ember-cli-update   "0.0.29",
  ember-cli-update   "0.0.32",
  ember-cli-update   "0.0.33",
  ember-cli-update   "0.0.34",
  ember-cli-update   "0.0.35",
  ember-cli-update   "0.0.36",
  ember-cli-update   "0.0.37",
  ember-cli-update   "0.0.38",
  ember-cli-update   "0.0.39",
  ember-cli-update   "0.0.40",
  ember-cli-update   "0.0.41",
  ember-cli-update   "0.0.42",
  ember-cli-update   "0.0.43",
  ember-cli-update   "0.0.44",
  ember-cli-update   "0.0.45",
  ember-cli-update   "0.0.46",
  ember-cli-update   "0.0.47",
  ember-cli-update   "0.1.0",
  ember-cli-update   "0.1.1",
  ember-cli-update   "0.1.2",
  ember-cli-update   "0.1.3",
  ember-cli-update   "0.1.4",
  ember-cli-update   "0.1.5",
  ember-cli-update   "0.1.6",
  ember-cli-update   "0.1.7",
  ember-cli-update   "0.1.8",
  ember-cli-update   "0.1.9",
  ember-cli-update   "0.1.10",
  ember-cli-update   "0.1.11",
  ember-cli-update   "0.1.12",
  ember-cli-update   "0.1.13",
  ember-cli-update   "0.1.14",
  ember-cli-update   "0.1.15",
  ember-cli-update   "0.2.0-beta.1",
  ember-cli-update   "0.2.0",
  ember-cli-update   "0.2.1",
  ember-cli-update   "0.2.2",
  ember-cli-update   "0.2.3",
  ember-cli-update   "0.2.4",
  ember-cli-update   "0.2.5",
  ember-cli-update   "0.2.6",
  ember-cli-update   "0.2.7",
  ember-cli-update   "1.13.0",
  ember-cli-update   "1.13.1",
  ember-cli-update   "1.13.5",
  ember-cli-update   "1.13.6",
  ember-cli-update   "1.13.7",
  ember-cli-update   "1.13.8",
  ember-cli-update   "1.13.9",
  ember-cli-update   "1.13.10",
  ember-cli-update   "1.13.11",
  ember-cli-update   "1.13.12",
  ember-cli-update   "1.13.13",
  ember-cli-update   "1.13.14",
  ember-cli-update   "1.13.15",
  ember-cli-update   "2.2.0-beta.1",
  ember-cli-update   "2.2.0-beta.2",
  ember-cli-update   "2.2.0-beta.3",
  ember-cli-update   "2.2.0-beta.4",
  ember-cli-update   "2.2.0-beta.5",
  ember-cli-update   "2.2.0-beta.6",
  ember-cli-update   "2.3.0-beta.1",
  ember-cli-update   "2.3.0-beta.2",
  ember-cli-update   "2.3.0",
  ember-cli-update   "2.4.0",
  ember-cli-update   "2.4.1",
  ember-cli-update   "2.4.2",
  ember-cli-update   "2.4.3",
  ember-cli-update   "2.5.0",
  ember-cli-update   "2.5.1",
  ember-cli-update   "2.6.0-beta.1",
  ember-cli-update   "2.6.0-beta.2",
  ember-cli-update   "2.6.0-beta.3",
  ember-cli-update   "2.6.0",
  ember-cli-update   "2.6.1",
  ember-cli-update   "2.6.2",
  ember-cli-update   "2.6.3",
  ember-cli-update   "2.7.0-beta.1",
  ember-cli-update   "2.7.0-beta.2",
  ember-cli-update   "2.7.0-beta.3",
  ember-cli-update   "2.7.0-beta.4",
  ember-cli-update   "2.7.0-beta.5",
  ember-cli-update   "2.7.0-beta.6",
  ember-cli-update   "2.7.0",
  ember-cli-update   "2.8.0-beta.1",
  ember-cli-update   "2.8.0-beta.2",
  ember-cli-update   "2.8.0-beta.3",
  ember-cli-update   "2.8.0",
  ember-cli-update   "2.9.0-beta.1",
  ember-cli-update   "2.9.0-beta.2",
  ember-cli-update   "2.9.0",
  ember-cli-update   "2.9.1",
  ember-cli-update   "2.10.0-beta.1",
  ember-cli-update   "2.10.0-beta.2",
  ember-cli-update   "2.10.0",
  ember-cli-update   "2.10.1",
  ember-cli-update   "2.11.0-beta.1",
  ember-cli-update   "2.11.0-beta.2",
  ember-cli-update   "2.11.0",
  ember-cli-update   "2.11.1",
  ember-cli-update   "2.12.0-beta.1",
  ember-cli-update   "2.12.0-beta.2",
  ember-cli-update   "2.12.0",
  ember-cli-update   "2.12.1",
  ember-cli-update   "2.12.2",
  ember-cli-update   "2.12.3",
  ember-cli-update   "2.13.0-beta.1",
  ember-cli-update   "2.13.0-beta.2",
  ember-cli-update   "2.13.0-beta.3",
  ember-cli-update   "2.13.0-beta.4",
  ember-cli-update   "2.13.0",
  ember-cli-update   "2.13.1",
  ember-cli-update   "2.13.2",
  ember-cli-update   "2.13.3",
  ember-cli-update   "2.14.0-beta.1",
  ember-cli-update   "2.14.0-beta.2",
  ember-cli-update   "2.14.0",
  ember-cli-update   "2.14.1",
  ember-cli-update   "2.14.2",
  ember-cli-update   "2.15.0-beta.1",
  ember-cli-update   "2.15.0-beta.2",
  ember-cli-update   "2.15.0",
  ember-cli-update   "2.15.1",
  ember-cli-update   "2.16.0-beta.1",
  ember-cli-update   "2.16.0-beta.2",
  ember-cli-update   "2.16.0",
  ember-cli-update   "2.16.1",
  ember-cli-update   "2.16.2",
  ember-cli-update   "2.17.0-beta.1",
  ember-cli-update   "2.17.0-beta.2",
  ember-cli-update   "2.17.0",
  ember-cli-update   "2.17.1",
  ember-cli-update   "2.17.2",
  ember-cli-update   "2.18.0-beta.1",
  ember-cli-update   "2.18.0-beta.2",
  ember-cli-update   "2.18.0",
  ember-cli-update   "2.18.1",
  ember-cli-update   "3.0.0-beta.1",
  ember-cli-update   "3.0.0-beta.2"
  ember-cli-update ]
  ember-cli-update  +475ms
  ember-cli-update npm info ember-cli@latest version --json +8ms
  ember-cli-update "2.18.1"
  ember-cli-update  +459ms
  git-diff-apply git status --porcelain +0ms
  git-diff-apply  +9ms
  git-diff-apply git clone --mirror https://github.com/ember-cli/ember-new-output /var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv/.git +1ms
Cloning into bare repository '/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv/.git'...
  git-diff-apply  +448ms
  git-diff-apply git --git-dir="/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv/.git" rev-parse v2.18.0 +1ms
  git-diff-apply 5490afeee3d374a496ca683080234eb6f2da3c27
  git-diff-apply  +7ms
  git-diff-apply git --git-dir="/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv/.git" --work-tree="/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv" checkout 5490afeee3d374a496ca683080234eb6f2da3c27 +0ms
Note: checking out '5490afeee3d374a496ca683080234eb6f2da3c27'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 5490afe 2.18.0
  git-diff-apply  +18ms
  git-diff-apply git --git-dir="/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv/.git" rev-parse v2.18.1 +3ms
fatal: ambiguous argument 'v2.18.1': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
{ Error: Command failed: git --git-dir="/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv/.git" rev-parse v2.18.1
fatal: ambiguous argument 'v2.18.1': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

    at checkExecSyncError (child_process.js:601:13)
    at Object.execSync (child_process.js:641:13)
    at run (/Users/james/.nvm/versions/node/v8.9.3/lib/node_modules/ember-cli-update/node_modules/git-diff-apply/src/run.js:8:19)
    at checkOutTag (/Users/james/.nvm/versions/node/v8.9.3/lib/node_modules/ember-cli-update/node_modules/git-diff-apply/src/check-out-tag.js:9:13)
    at buildReturnObject (/Users/james/.nvm/versions/node/v8.9.3/lib/node_modules/ember-cli-update/node_modules/git-diff-apply/src/index.js:69:5)
    at Promise.resolve.then (/Users/james/.nvm/versions/node/v8.9.3/lib/node_modules/ember-cli-update/node_modules/git-diff-apply/src/index.js:148:20)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
    at Function.Module.runMain (module.js:678:11)
    at startup (bootstrap_node.js:187:16)
  error: null,
  cmd: 'git --git-dir="/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv/.git" rev-parse v2.18.1',
  file: '/bin/sh',
  args:
   [ '/bin/sh',
     '-c',
     'git --git-dir="/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv/.git" rev-parse v2.18.1' ],
  options:
   { shell: true,
     file: '/bin/sh',
     args:
      [ '/bin/sh',
        '-c',
        'git --git-dir="/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/tmp-27377kfdoyle37jnv/.git" rev-parse v2.18.1' ],
     envPairs:
      [ 'TERM_SESSION_ID=w0t1p0:779FF531-8A65-4F32-97B9-08C69ADE5B70',
        'SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.crTCDN6ZKH/Listeners',
        'Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.6m8HLLAIno/Render',
        'COLORFGBG=7;0',
        'ITERM_PROFILE=Default',
        'XPC_FLAGS=0x0',
        'LANG=en_CA.UTF-8',
        'PWD=/Users/james/Projects/daily-bugle-app',
        'SHELL=/usr/local/bin/zsh',
        'TERM_PROGRAM_VERSION=3.1.5.beta.2',
        'TERM_PROGRAM=iTerm.app',
        'PATH=/Users/james/.nvm/versions/node/v8.9.3/bin:/Users/james/.rbenv/shims:/usr/local/bin:/usr/local/share/npm/bin:/usr/local/sbin:./node_modules/.bin:/Users/james/.ansible-env/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/james/.npm-global/bin::/Applications/Postgres.app/Contents/Versions/latest/bin:/Users/james/.fzf/bin',
        'DISPLAY=/private/tmp/com.apple.launchd.kX0drw1yKJ/org.macosforge.xquartz:0',
        'COLORTERM=truecolor',
        'TERM=xterm-256color',
        'HOME=/Users/james',
        'TMPDIR=/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/',
        'USER=james',
        'XPC_SERVICE_NAME=0',
        'LOGNAME=james',
        '__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x52',
        'ITERM_SESSION_ID=w0t1p0:779FF531-8A65-4F32-97B9-08C69ADE5B70',
        'SHLVL=1',
        'OLDPWD=/Users/james',
        'ZSH=/Users/james/.oh-my-zsh',
        'ZSH_THEME=kolo',
        'PAGER=less',
        'LESS=-R',
        'LC_CTYPE=en_US.UTF-8',
        'LSCOLORS=Gxfxcxdxbxegedabagacad',
        'ZSH_TMUX_TERM=screen-256color',
        '_ZSH_TMUX_FIXED_CONFIG=/Users/james/.oh-my-zsh/plugins/tmux/tmux.extra.conf',
        'EDITOR=nvim',
        'NODE_PATH=/usr/local/lib/node_modules:/Users/james/.npm-global',
        'RBENV_SHELL=zsh',
        'BASE16_THEME=base16-monokai',
        'FZF_DEFAULT_COMMAND=rg --files --follow --hidden',
        'NVM_DIR=/Users/james/.nvm',
        'NVM_CD_FLAGS=-q',
        'NVM_BIN=/Users/james/.nvm/versions/node/v8.9.3/bin',
        'ERL_AFLAGS=-kernel shell_history enabled',
        'DEBUG=ember-cli-update,git-diff-apply',
        '_=/Users/james/.nvm/versions/node/v8.9.3/bin/ember-cli-update' ],
     killSignal: undefined,
     stdio: [ [Object], [Object], [Object] ] },
  envPairs:
   [ 'TERM_SESSION_ID=w0t1p0:779FF531-8A65-4F32-97B9-08C69ADE5B70',
     'SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.crTCDN6ZKH/Listeners',
     'Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.6m8HLLAIno/Render',
     'COLORFGBG=7;0',
     'ITERM_PROFILE=Default',
     'XPC_FLAGS=0x0',
     'LANG=en_CA.UTF-8',
     'PWD=/Users/james/Projects/daily-bugle-app',
     'SHELL=/usr/local/bin/zsh',
     'TERM_PROGRAM_VERSION=3.1.5.beta.2',
     'TERM_PROGRAM=iTerm.app',
     'PATH=/Users/james/.nvm/versions/node/v8.9.3/bin:/Users/james/.rbenv/shims:/usr/local/bin:/usr/local/share/npm/bin:/usr/local/sbin:./node_modules/.bin:/Users/james/.ansible-env/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/james/.npm-global/bin::/Applications/Postgres.app/Contents/Versions/latest/bin:/Users/james/.fzf/bin',
     'DISPLAY=/private/tmp/com.apple.launchd.kX0drw1yKJ/org.macosforge.xquartz:0',
     'COLORTERM=truecolor',
     'TERM=xterm-256color',
     'HOME=/Users/james',
     'TMPDIR=/var/folders/bj/d0jc0q090k3bg9lwlpltyt6r0000gn/T/',
     'USER=james',
     'XPC_SERVICE_NAME=0',
     'LOGNAME=james',
     '__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x52',
     'ITERM_SESSION_ID=w0t1p0:779FF531-8A65-4F32-97B9-08C69ADE5B70',
     'SHLVL=1',
     'OLDPWD=/Users/james',
     'ZSH=/Users/james/.oh-my-zsh',
     'ZSH_THEME=kolo',
     'PAGER=less',
     'LESS=-R',
     'LC_CTYPE=en_US.UTF-8',
     'LSCOLORS=Gxfxcxdxbxegedabagacad',
     'ZSH_TMUX_TERM=screen-256color',
     '_ZSH_TMUX_FIXED_CONFIG=/Users/james/.oh-my-zsh/plugins/tmux/tmux.extra.conf',
     'RACKSPACE_API_KEY=f9b92f89fde3156aaceba38585b89415',
     'RACKSPACE_USERNAME=pnunmanaged',
     'EDITOR=nvim',
     'NODE_PATH=/usr/local/lib/node_modules:/Users/james/.npm-global',
     'RBENV_SHELL=zsh',
     'BASE16_THEME=base16-monokai',
     'FZF_DEFAULT_COMMAND=rg --files --follow --hidden',
     'NVM_DIR=/Users/james/.nvm',
     'NVM_CD_FLAGS=-q',
     'NVM_BIN=/Users/james/.nvm/versions/node/v8.9.3/bin',
     'ERL_AFLAGS=-kernel shell_history enabled',
     'DEBUG=ember-cli-update,git-diff-apply',
     '_=/Users/james/.nvm/versions/node/v8.9.3/bin/ember-cli-update' ],
  stderr: <Buffer 66 61 74 61 6c 3a 20 61 6d 62 69 67 75 6f 75 73 20 61 72 67 75 6d 65 6e 74 20 27 76 32 2e 31 38 2e 31 27 3a 20 75 6e 6b 6e 6f 77 6e 20 72 65 76 69 73 ... >,
  stdout: <Buffer 76 32 2e 31 38 2e 31 0a>,
  pid: 27389,
  output:
   [ null,
     <Buffer 76 32 2e 31 38 2e 31 0a>,
     <Buffer 66 61 74 61 6c 3a 20 61 6d 62 69 67 75 6f 75 73 20 61 72 67 75 6d 65 6e 74 20 27 76 32 2e 31 38 2e 31 27 3a 20 75 6e 6b 6e 6f 77 6e 20 72 65 76 69 73 ... > ],
  signal: null,
  status: 128 }
```

**Comment by Gaurav0** [comment]
TL;DR: fatal: ambiguous argument 'v2.18.1': unknown revision or path not in the working tree.

**Comment by Gaurav0** [comment]
Basically the problem is that `ember-new-output` has not yet been updated. The command needs to handle this error condition gracefully and not attempt the update.

**Comment by denzo** [comment]
Having the same issue 😢

**Comment by backspace** [comment]
@denzo, to clarify @Gaurav0’s comment, were you able to use `ember-cli-update --to 2.18.0` instead?

**Comment by kellyselden** [comment]
I fixed the output repos, so this shouldn't be a problem anymore. I will file a new issue to make the error more clear.
